exist-core/src/main/resources/org/exist/xquery/lib/xqsuite/xqsuite.xql

```diff
@@ -819,14 +909,27 @@ declare %private function test:assertTrue($result as item()*) as element(report)
  : Check if the function caused an error.
  :)
 declare %private function test:assertError($value as xs:string, $result as item()*) as element(report)? {
-    if ($result) then
-        ()
-    else
-        <report>
-            <failure message="assertError failed. Expected error {$value}"
-                type="failure-error-code-1"/>
-            <output>{ $result }</output>
-        </report>
+    let $ebv :=
```

Comment:
This code looks like it could be written in a more readable way.

Reply:
In which cases does effectiveBoolean value fail - and could that not be mitigated by a call to `exists()` ?

Reply:
```
declare %private function test:assertError($value as xs:string, $result as item()*) as element(report)? {
    if (exists($result)) then ()
    else
        <report>
            <failure message="assertError failed. Expected error {$value}"
                type="failure-error-code-1"/>
            <output>{ $result }</output>
        </report>
};
```
is sufficient. I just tested it. This change alone shows that the array:put test actually fails. Nothing else needed.

Reply:
`$result` is of type `item()*`, so in the previous code:

```
if ($result)
```

This is getting the *effective boolean value* of `$result`. That is not the same as your proposed `fn:exists($result)` which would instead just tell you if the sequence is non-empty.

Reply:
OK, I understand now. In that case it would help future readers of the source code to extract the evaluation of a safe effective boolean value into a properly named function.

```xquery
declare %private function test:safeEffectiveBooleanValue ($result as item()*) as xs:boolean {
    try {
        not(not(($result)))
    }
    catch * {
        false()
    }
};

```